go.mod

```diff
@@ -26,14 +28,14 @@ require (
 	go.opentelemetry.io/otel/sdk/metric v1.35.0
 	go.uber.org/zap v1.27.0
 	gopkg.in/yaml.v2 v2.4.0
-	k8s.io/api v0.29.4
```

Comment:
Could we split the Kubernetes version bump in a separate PR? https://github.com/grafana/tempo-operator/pull/1157 was already created by dependabot. This would make this PR a bit smaller.

Also, we need to do a few things for Kubernetes version bumps: update GitHub actions, add `kind-*.yml` file, and add a new changelog entry.